langchain4j/src/main/java/dev/langchain4j/service/DefaultAiServices.java

```diff
@@ -351,6 +384,21 @@ public Object invoke(Object proxy, Method method, Object[] args) throws Throwabl
                         }
                     }
 
+                    private Optional<InvocationParameters> findInvocationParameters(Object[] arguments, Parameter[] parameters) {
+                        if (arguments == null) {
+                            return Optional.empty();
+                        }
+                        for (int i = 0; i < parameters.length; i++) {
+                            Parameter parameter = parameters[i];
+                            if (parameter.getType().isAssignableFrom(InvocationParameters.class)) {
```

Comment:
I always get confused about this, but I believe that this should be the opposite:

`if (InvocationParameters.class.isAssignableFrom(parameter.getType()))`

or even better maybe just use

`if (parameter.getType() instanceof InvocationParameters)`

that imo is more readable and less error-prone.